src/coordinator/mod.rs

```diff
@@ -409,6 +413,47 @@ impl Coordinator {
         Ok(())
     }
 
+    // Unlike input registers, holding registers are not broadcast by inverters,
+    // but they are interesting nevertheless. Broadcasting the holding registers
+    // when we connect to an inverter makes it easy for configuration data to be
+    // tracked, which is particularly useful in conjunction with HomeAssistant.
+    async fn inverter_connected(&self, datalog: Serial) -> Result<()> {
+        let inverter = match self.config.enabled_inverter_with_datalog(datalog) {
+            Some(inverter) => inverter,
+            None => bail!("Unknown inverter connected: {}", datalog),
+        };
+
+        if !inverter.broadcast_holding_registers() {
+            return Ok(());
+        }
+
+        info!("Reading holding registers for inverter {}", datalog);
+
+        // We can only read holding registers in blocks of 40. Reading 41 bytes
+        // results in a short packet back from the inverter (error response?)
+        let mut register = 0;
+        while register < lxp::packet::NUM_HOLDING_REGISTERS {
+            let count = std::cmp::min(40, lxp::packet::NUM_HOLDING_REGISTERS - register);
+
+            self.read_hold(inverter.clone(), register as i16, count)
```

Comment:
I haven't bothered to look at making these retained messages yet.

Reply:
I have some WIP for this and it makes home assistant much nicer to use, but I've got too much in-flight just now :sweat_smile:. I'll submit it inabit.